### ISSUE TITLE
您好，可以向您请教问题吗？

### ISSUE THREAD

**Issue by yujunhuics** [tdt98/Multi-Grained-Named-Entity-Recognition-Improvement-with-BERT-and-Focal-Loss#1]
bert的hdf5是如何生成的呢？

**Comment by tdt98** [comment]
I used HuggingFace pretrained BERT model to generate BERT hdf5 file.
This is the reference https://huggingface.co/transformers/model_doc/bert.html#overview.
Thanks for asking!

**Comment by yujunhuics** [comment]
Thank you very much. When Elmo generates HDF5, what is the original data format? generate elmo.hdf5 Can you give me some guidance? I want to run your code as soon as possible? Learn your method. Thank you very much for your guidance!
非常感谢，elmo生成hdf5时，原始的数据格式是怎样的？生成elmo.hdf5的方法您能指导一下吗？我想尽快的将您的代码跑起来？学习一下您的方法。如能得到您的指导，感激不尽！

**Comment by tdt98** [comment]
The format of input is:
Xinhua News Agency , Canberra , August 31st	NNP NNP NNP , NNP , NNP NN	2 0 2	3 4 4
Tokenized setenece \t POS tag \t position of entities
2 0 2 means: 
2 is the type of entity, 0 is the starting position of entity, 2 is the ending position of entity.
This original code you can reference at :
https://github.com/congyingxia/Multi-Grained-NER
I just upgrade original model by replacing BERT by ELMo and CrossEntropy by Focal Loss in Detector sub-components

**Comment by yujunhuics** [comment]
`
file=$1
allennlp elmo sentences/${file}_sentences elmo_${file}.hdf5 --all

nohup sh dump_feature.sh test > test.log 2>&1 &
nohup sh dump_feature.sh train > train.log 2>&1 &
nohup sh dump_feature.sh dev > dev.log 2>&1 &

`
Thank you very much, but according to these two scripts, it seems that they can't be generated elmo.hdf5 Documents? Because of the following errors:

`
usage: allennlp [-h] [--version]  ...
allennlp: error: argument : invalid choice: 'elmo' (choose from 'build-vocab', 'cached-path', 'evaluate', 'find-lr', 'predict', 'print-results', 'test-install', 'train')
`

I want to ask you that it is also generated according to this script elmo.hdf5 Documents? Is there any other way?

**Comment by tdt98** [comment]
I never met those errors when running the original code. 
The code was good with me.
 You can check the version of AllenNLP again.
My AllenNLP version is 0.9.0 and Pytorch version is 1.2.0
You can find another pretrained ELMo model if it does not work.
Hope everything is fine with you.

**Comment by yujunhuics** [comment]
Ok,thank you very much for your valuable time to guide me patiently. I wish you a happy life

**Comment by tdt98** [comment]
You are welcome. Feel free to ask me if you have any questions.

**Comment by yujunhuics** [comment]
Thank you very much for your advice. The HDF5 file was successfully generated. It turns out that it is really the allennlp version problem.

**Comment by yujunhuics** [comment]
I would like to ask you, for the original author's data set processing, is nltk annotated?

**Comment by tdt98** [comment]
I am sorry, this question is out of my knowledge. You can comment in the original author's repository for answer.

**Comment by yujunhuics** [comment]
It doesn't matter. Thank you all the same
